### PR TITLE
chore: paint it black and sand ruff edges

### PR DESCRIPTION
**Pull request by dmaljovec**
# What?
 - Run `black` against this repository
 - Fix `ruff .` errors
 - Configure `tox` and `pre-commit` to use `black` in place of `yapf` and `ruff` in place of `flake8`

# Why?
 - Our internal company standard has recently been switched to `black`

### CODE REVIEW

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1 @@
+48819da8bb9d1b7fbc2911c3865f2188ee0400bb
```
Comment (dmaljovec) [inline .git-blame-ignore-revs:1]: @bengioe once we squash and merge this, we will update this value to the sha of that merge commit.

File: pyproject.toml
Comment (dmaljovec) [inline pyproject.toml:1]: My script was a little aggressive against this file, let me try to reconcile this.

File: pyproject.toml [hunk 2]
```diff
@@ -1,69 +1,31 @@
 [build-system]
-requires = ["setuptools", "setuptools-scm", "wheel"]
+requires = ["setuptools", "wheel", "roadie"]
```
Comment (dmaljovec) [inline pyproject.toml]:
```suggestion
requires = ["setuptools", "setuptools-scm", "wheel"]
```

File: pyproject.toml [hunk 3]
```diff
@@ -2,68 +2,30 @@
...
 
 [tool.isort]
-profile = "google"
-line_length = 120
+profile = "black"
 py_version = "auto"
+line_length = 120
+
+[tool.bandit]
+exclude = "tests"
```
Comment (dmaljovec) [inline pyproject.toml]:
```suggestion
[tool.bandit]
skips = ["B101"]
exclude_dirs = ["tests", ".tox", ".venv"]
```

File: pyproject.toml [hunk 4]
```diff
@@ -2,68 +2,30 @@
...
-# 	"Programming Language :: Python",
-# 	"Programming Language :: Python :: 3"
-# ]
-# keywords = ["gflownet"]
-# requires-python = ">=3.6"
-# readme = "README.md"
-
-[tool.bandit]
-skips = ["B101"]
-exclude_dirs = ["tests", ".tox", ".venv"]
```
Comment (dmaljovec) [inline pyproject.toml]: re-added below

File: pyproject.toml [hunk 5]
```diff
@@ -2,68 +2,30 @@
...
-
-# [tool.flake8] not yet supported
-# max-line-length = 120
-# exclude = ".github,doc,.git,.nox,.tox,doc"
-# max_complexity = 20
```
Comment (dmaljovec) [inline pyproject.toml:27]: Do you want to switch to `ruff`?

Comment (bengioe) [reply pyproject.toml:27]: Yes, might as well :)

File: pyproject.toml [hunk 6]
```diff
@@ -2,68 +2,31 @@
...
-[[tool.mypy.overrides]]
-module = "tests.*"
-allow_untyped_defs = true
-allow_incomplete_defs = true
-
-[tool.isort]
-profile = "google"
-line_length = 120
-py_version = "auto"
+[tool.black]
```
Comment (dmaljovec) [inline pyproject.toml:45]:
```suggestion
[tool.pytest.ini_options]
addopts = [
    "-v",
    "-x",
    "--color=yes",
    "--cov-report=term-missing",
    "--cov=gflownet",
    "--typeguard-packages=ml_kit,tests"
]
testpaths = ["tests"]
pythonpath = "src/"

[tool.black]
```

File: pyproject.toml [hunk 7]
```diff
@@ -2,39 +2,19 @@
...
-# author_email = "devs@recursionpharma.com"
-# classifiers = [
-# 	"Operating System :: OS Independent",
-# 	"Programming Language :: Python",
-# 	"Programming Language :: Python :: 3"
-# ]
-# keywords = ["gflownet"]
-# requires-python = ">=3.6"
-# readme = "README.md"
+[tool]
```
Comment (dmaljovec) [inline pyproject.toml]:
```suggestion
```

File: pyproject.toml [hunk 8]
```diff
@@ -47,23 +26,18 @@ addopts = [
...
-[[tool.mypy.overrides]]
-module = "tests.*"
-allow_untyped_defs = true
-allow_incomplete_defs = true
-
-[tool.isort]
-profile = "google"
-line_length = 120
-py_version = "auto"
+[tool.black]
```
Comment (dmaljovec) [inline pyproject.toml:45]:
```suggestion
[tool.mypy]
ignore_missing_imports = true
show_error_codes = true
show_error_context = true
show_traceback = true
strict = false
strict_optional = false
implicit_reexport = true
allow_redefinition = true
files = "src"

[[tool.mypy.overrides]]
module = "tests.*"
allow_untyped_defs = true
allow_incomplete_defs = true

[tool.black]
```

File: pyproject.toml [hunk 9]
```diff
@@ -2,39 +2,18 @@
...
-# keywords = ["gflownet"]
-# requires-python = ">=3.6"
-# readme = "README.md"
+[tool.distutils.bdist_wheel]
+universal = "true"
+
+[tool.isort]
+profile = "black"
+py_version = "auto"
+line_length = 120
```
Comment (dmaljovec) [inline pyproject.toml]:
```suggestion

```

File: pyproject.toml [hunk 10]
```diff
@@ -63,7 +42,18 @@ module = "tests.*"
 allow_untyped_defs = true
 allow_incomplete_defs = true
 
-[tool.isort]
-profile = "google"
-line_length = 120
-py_version = "auto"
+[tool.black]
```
Comment (dmaljovec) [inline pyproject.toml:45]:
```suggestion
[tool.isort]
profile = "black"
py_version = "auto"
line_length = 120

[tool.black]
```

**Comment by bengioe** [comment]
I guess the `flake8` step should be removed/deactivated in `python/library`?

**Comment by dmaljovec** [comment]
> I guess the `flake8` step should be removed/deactivated in `python/library`?

Oops! I forgot we don't have support for that yet. I'll get that added.

**Comment by dmaljovec** [comment]
Here is a PR that should enable that behavior for our internal pipeline: https://github.com/recursionpharma/build-pipelines/pull/1142

**Comment by dmaljovec** [comment]
Green again 😌

**Review by bengioe — APPROVED** [review]

**Pull request merged**